test/integration/hotrestart_handoff_test.py

```diff
@@ -0,0 +1,255 @@
+"""Tests the behavior of connection handoff between instances during hot restart.
+
+Specifically, tests that:
+1. A tcp connection opened before hot restart begins continues to function during drain.
+2. A tcp connection opened after hot restart begins while the old instance is still running
+   goes to the new instance.
+TODO(ravenblack): perform the same tests for quic connections once they will work as expected.
+"""
+
+from datetime import datetime
+import json
+import os
+import random
+import unittest
+from aiohttp import client_exceptions, web, web_runner, ClientSession
+import asyncio
+import logging
+import subprocess
+import sys
+import threading
+
+def random_loopback_host():
+    """Returns a randomized loopback IP.
+    This can be used to reduce the chance of port conflicts when tests are
+    running in parallel."""
+    return f"127.{random.randrange(0,256)}.{random.randrange(0,256)}.{random.randrange(1, 255)}"
```

Comment:
Yes. Is there any way to *not* be flaky on port-already-in-use for integration tests though? (Google used a whole mess of infrastructure and awkward code to select non-conflicting ports for tests running in their test-infra, and it was still significantly flakier than this will be!)